### PR TITLE
Migrate /upload and /upload-key to app router

### PR DESCRIPTION
**Pull request by francis-robert**
✨ feat(upload-key): add new route and components for upload key management

The new route and components allow users to manage their upload keys, which are used to automatically upload splits from LiveSplit. Users can now view, copy, and use their upload keys, as well as access instructions on how to set up the LiveSplit component and troubleshoot any issues. This feature enhances the user experience by making it easier to manage and use upload keys.

✨ feat(upload/page.tsx): add new Upload page with drag and drop functionality 🔧 refactor(session.action.ts): update getSession return type to Promise<User> for better type checking 🔥 remove(api/users/[user]/upload-key.ts): delete unused upload-key API endpoint The new Upload page allows users to upload their splits using a drag and drop interface. The getSession function in session.action.ts has been updated to return a Promise<User> for better type checking and improved code quality. The unused upload-key API endpoint has been removed to keep the codebase clean and maintainable.

### CODE REVIEW

File: app/upload-key/page.tsx [hunk 1]
```diff
@@ -1,83 +1,36 @@
...
-    const [showUploadKey, setShowUploadKey] = useState(false);
-    const [copySuccess, setCopySuccess] = useState(false);
-
-    const { data, error } = useSWR(
-        `/api/users/${session.id}-${session.username}/upload-key`,
-        fetcher
+export default async function UploadKey() {
+    const session = await getSession();
+    const baseUrl = getBaseUrl();
+    const data = await fetch(
```
Comment (therungg) [inline app/upload-key/page.tsx:10]: What happens now if this errors? Does it just give a generic error page?

Not too familiar with the switch from useSWR to regular fetch() yet

File: app/upload-key/copy-upload-key.component.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+const CopyUploadKeyButton: React.FunctionComponent<
+    CopyUploadKeyButtonProps
+> = ({ uploadKey, isCopied, setIsCopied }) => {
+    return (
+        <div className={uploadKeyStyles.uploadKeyBox}>
+            {uploadKey}
+            <div
+                className={uploadKeyStyles.uploadKey}
+                onClick={async () => {
+                    await navigator.clipboard.writeText(uploadKey);
```
Comment (therungg) [inline app/upload-key/copy-upload-key.component.tsx:42]: Didn't even know this needed to be async, it always worked without awaiting lol

**Review by therungg — APPROVED** [review]
one question, but feel free to resolve convo's and merge

File: app/upload-key/page.tsx (hunk 1 above)
Comment (francis-robert) [reply app/upload-key/page.tsx:10]: Good call! As discussed offline, I added an error.tsx handler as a generic catch-all for now

**Review by therungg — APPROVED** [review]
Awesome <3

**Pull request merged**